File: Algorithm/merge_intervals.py
class Interval:
    def __init__ (self, s=0, e=0):
        self.start = s
        self.end = e 
class Solution:
    def merge(self,intervals):
        res=[]
        intervals.sort(key = lambda x:x.start)
        res.append(intervals[0])
        n = len(intervals)
        for index in range(1, n):
            if res[-1].end <intervals[index].start:
                res.append(intervals[index])
            else:
                if res[-1].end < intervals[index].end:
                    res[-1].end  = intervals[index].end
        return res
                    
                                    
if __name__=='__main__':
    sol=Solution()
    lsts=[[1,4],[2,5]]
    intervals = []
    for lst in lsts:
  
        interval = Interval(lst[0], lst[1])

        intervals.append(interval)
        
    for lst in  sol.merge(intervals):
        print lst.start, lst.end

File: Algorithm/CPP/merge_intervals.cpp
# include<stdio.h>
# include <iostream>
# include <string>
# include <array>
# include <vector>
using namespace std;


struct Interval {
    int start;
    int end;
    Interval() : start(0), end(0) {}
    Interval(int s, int e) : start(s), end(e) {}
};
class Solution {
public:
	static bool cmp(Interval left, Interval right)
	{
		return left.start < right.start;
	}
	vector<Interval> merge(vector<Interval>& intervals) 
	{
		sort(intervals.begin(), intervals.end(), cmp);
		vector<Interval> res;
		if (intervals.empty()) return res;
		res.push_back(intervals[0]);
		for (int i = 1; i < intervals.size(); ++i)
		{
			if (intervals[i].start > res.back().end)
			{
				res.push_back(intervals[i]);
			} else
			{
				res.back().end = max(res.back().end,intervals[i].end);
			}
		}
		return res;
	}
};


int main()
{
	vector<Interval> intervals = { Interval(1, 3), Interval(2, 6), Interval(8, 10), Interval(15, 18) };
	Solution sol;
	for (auto k : sol.merge(intervals))
	{
		cout << "(" << k.start << ","<<k.end << ")" << endl;
	}
	system("pause");
}

